test/cancel_test.go

```diff
@@ -82,29 +82,22 @@ func TestTaskRunPipelineRunCancel(t *testing.T) {
 	}
 
 	t.Logf("Waiting for TaskRuns from PipelineRun %s in namespace %s to be running", "pear", namespace)
-	errChan := make(chan error, len(taskrunList.Items))
-	defer close(errChan)
-
 	for _, taskrunItem := range taskrunList.Items {
-		go func(chan error) {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```

Comment:
```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop